server/src/main/java/org/opensearch/cluster/decommission/DecommissionService.java

```diff
@@ -303,19 +309,66 @@ public void onFailure(Exception e) {
+        List<String> awarenessValues = forcedAwarenessAttributes.get(decommissionAttribute.attributeName());
+
+        decommissionController.updateMetadataWithDecommissionStatus(DecommissionStatus.WEIGH_AWAY, new ActionListener<>() {
+            @Override
+            public void onResponse(DecommissionStatus status) {
+                logger.info("updated the decommission status to [{}]", status);
+                // set the weights
+
+                decommissionController.setWeightForDecommissionedZone(awarenessValues, new ActionListener<>() {
+                    @Override
+                    public void onResponse(ClusterPutWRRWeightsResponse response) {
+                        clearVotingConfigExclusionAndUpdateStatus(true, true);
+                        failDecommissionedNodes(state, timeOutForNodeDraining);
+                    }
+
+                    @Override
+                    public void onFailure(Exception e) {
+                        clearVotingConfigExclusionAndUpdateStatus(false, false);
+                    }
+                });
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                logger.error(
+                    () -> new ParameterizedMessage(
+                        "failed to update decommission status for attribute [{}] to [{}]",
+                        decommissionAttribute.toString(),
+                        DecommissionStatus.WEIGH_AWAY
+                    ),
+                    e
+                );
+                // since we are not able to update the status, we will clear the voting config exclusion we have set earlier
+                clearVotingConfigExclusionAndUpdateStatus(false, false);
+            }
+        });
+    }
+
+    private void failDecommissionedNodes(ClusterState state, TimeValue timeOutForNodeDraining) {
+        // this method ensures no matter what, we always exit from this function after clearing the voting config exclusion
+        DecommissionAttributeMetadata decommissionAttributeMetadata = state.metadata().decommissionAttributeMetadata();
+        DecommissionAttribute decommissionAttribute = decommissionAttributeMetadata.decommissionAttribute();
+
         decommissionController.updateMetadataWithDecommissionStatus(DecommissionStatus.IN_PROGRESS, new ActionListener<>() {
             @Override
             public void onResponse(DecommissionStatus status) {
                 logger.info("updated the decommission status to [{}]", status);
                 // execute nodes decommissioning
-                decommissionController.removeDecommissionedNodes(
+
+                decommissionController.handleNodesDecommissionRequest(
                     filterNodesWithDecommissionAttribute(clusterService.getClusterApplierService().state(), decommissionAttribute, false),
                     "nodes-decommissioned",
                     TimeValue.timeValueSeconds(120L),
+                    timeOutForNodeDraining,
```

Comment:
node removal method should be independent of draining

Reply:
Done.